components/class-m-chart-admin.php

```diff
@@ -237,23 +232,38 @@ public function current_screen( $screen ) {
 			wp_enqueue_script(
 				'm-chart-admin',
 				$this->plugin_url . '/components/js/m-chart-admin.js',
-				array( 'highcharts', 'jquery' ),
+				array( 'jquery' ),
 				m_chart()->version
 			);
 
-			$settings = m_chart()->get_settings();
-			$post_id  = isset( $_GET['post'] ) ? absint( $_GET['post'] ) : '';
+			// Only load this if we are on an appropriate post page
+			$library = isset( $_GET['post'] ) ? m_chart()->get_post_meta( $_GET['post'], 'library' ) : m_chart()->get_library();
```

Comment:
I'd suggest casting the post variable as an int here and also switching from `isset()` to `empty()` so 0 isn't passed in to `get_post_meta()`.

```php
$library = empty( $_GET['post'] ) ? m_chart()->get_library() : m_chart()->get_post_meta( absint( $_GET['post'] ), 'library' );
```

Since those are long function names, perhaps just switching to an if/else would be easier to read?

```php
if ( empty( $_GET['post'] ) ) {
  $library = m_chart()->get_library();
} else {
  $library = m_chart()->get_post_meta( absint( $_GET['post'] ), 'library' );
}
```

Reply:
By casting as an int you men something like:

`php
$post_id = (int) $_GET['post'];
`

Stupid question probably but had to ask.